src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -252,6 +256,30 @@ describe('HeatmapPlot', () => {
     expect(component.find(Empty).length).toEqual(1);
   });
 
+  it('Shows loader if cell sets is empty', () => {
+    const store = mockStore({
+      ...initialState,
+      cellSets: {
+        ...initialState.cellSets,
+        hierarchy: [],
+        properties: [],
+        loading: false,
+        error: true,
+      },
+    });
+
+    component = mount(
+      <Provider store={store}>
+        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
+      </Provider>,
+    );
+
+    expect(component.find('VegaHeatmap').length).toEqual(0);
+
+    // 2 Because the Loader component contains ClipLoader, which contains a Loader
```

Comment:
just noticed in the code below, but if cellSets are empty, we should be displaying an empty component, not loading component

Reply:
Fixed